Robot2018/src/org/usfirst/frc/team199/Robot2018/subsystems/Drivetrain.java

```diff
@@ -42,6 +49,32 @@ public void initDefaultCommand() {
 		setDefaultCommand(new TeleopDrive());
 	}
 
+	/**
+	 * Sets the left side of the drivetrain to use the talons' pids to run at the
+	 * specified speed.
+	 * 
+	 * @param value
+	 *            The speed to run at
+	 */
+	public void dtLeftPIDDrive(double value) {
+		double setValue = value * Robot.getConst("Units per 100ms", 3413);
+		dtLeftMaster.set(ControlMode.Velocity, setValue);
+		dtLeftSlave.set(ControlMode.Velocity, setValue);
+	}
+
+	/**
+	 * Sets the right side of the drivetrain to use the talons' pids to run at the
+	 * specified speed.
+	 * 
+	 * @param value
+	 *            The speed to run at
+	 */
+	public void dtRightPIDDrive(double value) {
+		double setValue = value * Robot.getConst("Units per 100ms", 3413);
+		dtRightMaster.set(ControlMode.Velocity, setValue);
+		dtRightSlave.set(ControlMode.Velocity, setValue);
+	}
+
```

Comment:
This sort of duplication shouldn't be needed. Instead, move this method to a subclass of VelocityPIDController or create a common interface (that extends SpeedController) for both implementations. That class also needs to set the PID constants on the Talon.